internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -162,11 +173,17 @@ func (connector *SingleUseConnector) Connect(
 	// Create a channel for receiving the prelogin response through the context
 	preLoginResponseChannel := make(chan map[uint8][]byte)
 
-	// Create a channel for send the client login to the driver through the context
+	// Create a channel for sending the client login to the driver through the context
 	clientLoginChannel := make(chan mssql.Login)
 
+	// Create a channel for receiving a server error from the driver through the context
+	errChannel := make(chan mssql.Error)
+
+	// Create a channel for receiving a server loginAck from the driver through the context
+	loginAckChannel := make(chan mssql.LoginAckStruct)
+
 	// Set a 'marker' for when the driver has finished connecting to the server
-	connectPhaseFinished := make(chan struct{})
+	connectPhaseFinished := make(chan connectResult)
```

Comment:
We've slightly modified the function of this channel (which I like), so we might need to give it a more appropriate name. Maybe connectPhaseResponseChannel or connectionResultChannel?